### PR TITLE
Migrate to sbt-typelevel

### PR DESCRIPTION
**Pull request by armanbilge**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -42,13 +44,6 @@ jobs:
           distribution: temurin
           java-version: 8
 
-      - name: Setup Java (temurin@17)
```
Comment (osleonard) [inline .github/workflows/ci.yml:45]: Is this removal intentional ?

Comment (armanbilge) [reply .github/workflows/ci.yml:45]: Oh apologies, you were testing on 17 here as well. Good catch, easy to add back.
